core/state/statedb.go

```diff
@@ -636,6 +646,10 @@ func (s *StateDB) createObject(addr common.Address) (newobj, prev *stateObject)
 	} else {
 		s.journal.append(resetObjectChange{prev: prev, prevdestruct: prevdestruct})
 	}
+
+	newobj.created = true
+	newobj.sendalled = false
```

Comment:
Is it 100% certain that the `newobj` is flushed, so that when the next transaction is processed, we no longer have any objects with `created` set to true? I am not 100% certain, off the top of my head. 

(also, settings `sendalled` to false is not necessary, but it might be nice to do so anyway, as you have done)